cf-java-logging-support-logback/src/test/java/com/sap/hcp/cf/logging/common/TestCustomFields.java

```diff
@@ -14,101 +13,88 @@
 
-        assertThat(getMessage(), is(messageWithKeyValue));
-        assertThat(getCustomField(SOME_KEY), is(SOME_VALUE));
-    }
+		assertThat(getMessage(), is(TEST_MESSAGE));
+		assertThat(getCustomField(CUSTOM_FIELD_KEY), hasCustomField(CUSTOM_FIELD_KEY, SOME_VALUE, CUSTOM_FIELD_INDEX));
+	}
 
-    @Test
-    public void testEscape() {
-        String messageWithPattern = TEST_MESSAGE + " {}";
-        String strangeCharacters = "}{:\",\"";
-        String messageWithKeyValue = TEST_MESSAGE + " " + SOME_KEY + "=" + strangeCharacters;
+	@Test
+	public void testCustomFieldAsPartOfMessage() throws Exception {
+		String messageWithPattern = TEST_MESSAGE + " {}";
+		String messageWithKeyValue = TEST_MESSAGE + " " + CUSTOM_FIELD_KEY + "=" + SOME_VALUE;
 
-        LOGGER.info(messageWithPattern, customField(SOME_KEY, strangeCharacters));
+		LOGGER.info(messageWithPattern, customField(CUSTOM_FIELD_KEY, SOME_VALUE));
 
-        assertThat(getMessage(), is(messageWithKeyValue));
-        assertThat(getCustomField(SOME_KEY), is(strangeCharacters));
-    }
+		assertThat(getMessage(), is(messageWithKeyValue));
+		assertThat(getCustomField(CUSTOM_FIELD_KEY), hasCustomField(CUSTOM_FIELD_KEY, SOME_VALUE, CUSTOM_FIELD_INDEX));
+	}
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testNullKey() {
-        customField(null, SOME_VALUE);
-    }
+	@Test
+	public void testEscape() throws Exception {
+		String messageWithPattern = TEST_MESSAGE + " {}";
+		String messageWithKeyValue = TEST_MESSAGE + " " + CUSTOM_FIELD_KEY + "=" + HACK_ATTEMPT;
 
-    @Test
-    public void testNullValue() {
-        LOGGER.info(TEST_MESSAGE, customField(SOME_KEY, null));
+		LOGGER.info(messageWithPattern, customField(CUSTOM_FIELD_KEY, HACK_ATTEMPT));
 
-        assertThat(getMessage(), is(TEST_MESSAGE));
-        assertThat(getCustomField(SOME_KEY), is("null"));
-    }
+		assertThat(getMessage(), is(messageWithKeyValue));
+		assertThat(getCustomField(CUSTOM_FIELD_KEY),
+				hasCustomField(CUSTOM_FIELD_KEY, HACK_ATTEMPT, CUSTOM_FIELD_INDEX));
+	}
 
-    @Test
-    public void testLogMessageWithTwoCustomFields() {
-        LOGGER.info(TEST_MESSAGE, customField(SOME_KEY, SOME_VALUE), customField(SOME_OTHER_KEY, SOME_OTHER_VALUE));
+	@Test(expected = IllegalArgumentException.class)
+	public void testNullKey() {
+		customField(null, SOME_VALUE);
+	}
 
-        assertThat(getMessage(), is(TEST_MESSAGE));
+	@Test
+	public void testNullValue() throws Exception {
+		LOGGER.info(TEST_MESSAGE, customField(CUSTOM_FIELD_KEY, null));
 
-        assertThat(getCustomField(SOME_KEY), is(SOME_VALUE));
-        assertThat(getCustomField(SOME_OTHER_KEY), is(SOME_OTHER_VALUE));
-    }
+		assertThat(getMessage(), is(TEST_MESSAGE));
+		assertThat(getCustomField(CUSTOM_FIELD_KEY), hasCustomField(CUSTOM_FIELD_KEY, "null", CUSTOM_FIELD_INDEX));
+	}
 
-    @Test
-    public void testOrderOfLogMessageWithTwoCustomFields() {
```

Comment:
Why do we no longer need a test to check if the right number is assigned to a custom field when they are reported in inverse order while writing the log?

Reply:
This test was not very useful. `stringContainsInOrder` only tests, that the strings appear in that order. It does not verify, that the belong to the same custom field. So if the values were exchanged this test would still have been green. Furthermore, the order of the custom field entries is of no concern. Notice, that the new test actually does check, that the correct index `I` is created in the custom field entries.